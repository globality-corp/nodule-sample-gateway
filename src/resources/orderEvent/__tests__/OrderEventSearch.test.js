import request from 'supertest';
import { Nodule } from '@globality/nodule-config';
import { mockResponse } from '@globality/nodule-openapi';
import createApp from '../../../app';

let app;


beforeEach(async () => {
    await Nodule.testing().fromObject(
        mockResponse('charmander', 'order.search', {
            count: 1,
            items: [
                {
                    id: '00000000-0000-0000-0000-000000000000',
                    customerId: '00000000-0000-0000-0000-111111111111',
                },
            ],
        }),
    ).fromObject(
        mockResponse('charmander', 'orderEvent.search', {
            count: 2,
            items: [
                {
                    id: '00000000-0000-0000-0000-000000000000',
                    customerId: '00000000-0000-0000-0000-111111111111',
                },
                {
                    id: '00000000-0000-0000-0000-000000000000',
                    customerId: '00000000-0000-0000-0000-111111111112',
                },
            ],
        }),
    ).load();

    app = createApp();
});


it('should retrieve a list of events for each order', async () => {
    const query = `
        query {
          order {
            items {
                id
                customerId
                events {
                    items {
                        id
                        eventType
                    }
                }
            }
          }
        }
    `;

    const result = await request(app).post(
        '/gql/graphql',
    ).send({
        query,
    });

    expect(result.body.errors).toBeUndefined();
    expect(result.body).toMatchSnapshot();
});
