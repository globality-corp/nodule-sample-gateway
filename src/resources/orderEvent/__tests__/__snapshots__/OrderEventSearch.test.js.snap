// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should retrieve a list of events for each order 1`] = `
Object {
  "data": Object {
    "order": Object {
      "items": Array [
        Object {
          "customerId": "00000000-0000-0000-0000-111111111111",
          "events": Object {
            "items": Array [
              Object {
                "eventType": null,
                "id": "00000000-0000-0000-0000-000000000000",
              },
              Object {
                "eventType": null,
                "id": "00000000-0000-0000-0000-000000000000",
              },
            ],
          },
          "id": "00000000-0000-0000-0000-000000000000",
        },
      ],
    },
  },
}
`;
